src/ini.zig

```diff
@@ -0,0 +1,1204 @@
+        defer this.prop_idx += 1;
+
+        const prop = this.config.properties.ptr[this.prop_idx];
+
+        if (prop.key) |keyexpr| {
+            if (keyexpr.asUtf8StringLiteral()) |key| {
+                if (bun.strings.hasPrefixComptime(key, "@") and bun.strings.endsWith(key, ":registry")) {
+                    if (!this.count) {
+                        return .{
+                            .some = .{
+                                .scope = key[1 .. key.len - ":registry".len],
+                                .registry = brk: {
+                                    if (prop.value) |value| {
+                                        if (value.asUtf8StringLiteral()) |str| {
+                                            var parser = bun.Schema.Api.NpmRegistry.Parser{
+                                                .log = this.log,
+                                                .source = this.source,
+                                                .allocator = this.allocator,
+                                            };
+                                            break :brk parser.parseRegistryURLStringImpl(str) catch |e| {
+                                                if (e == error.OutOfMemory) bun.outOfMemory();
+                                                return error.ParserError;
+                                            };
+                                        }
+                                    }
+                                    return .none;
+                                },
+                            },
+                        };
+                    }
+
+                    return .none;
+                }
+            }
+        }
+
+        return .none;
+    }
+};
+
+pub fn loadNpmrcFromFile(
+    allocator: std.mem.Allocator,
+    install: *bun.Schema.Api.BunInstall,
+    env: *bun.DotEnv.Loader,
+    auto_loaded: bool,
+    log: *bun.logger.Log,
+) !void {
+    const npmrc_file = switch (bun.sys.openat(bun.FD.cwd(), ".npmrc", bun.O.RDONLY, 0)) {
+        .result => |fd| fd,
+        .err => |err| {
+            if (auto_loaded) return;
+            Output.prettyErrorln("{}\nwhile opening .npmrc \"{s}\"", .{
+                err,
+                ".npmrc",
+            });
+            Global.exit(1);
+        },
+    };
+    defer _ = bun.sys.close(npmrc_file);
+
+    var npmrc_contents = std.ArrayList(u8).init(allocator);
+    defer npmrc_contents.deinit();
+    switch (bun.sys.File.readToEndWithArrayList(bun.sys.File{ .handle = npmrc_file }, &npmrc_contents)) {
+        .result => {},
+        .err => |err| {
+            // TODO: should this exit(1)?
+            Output.prettyErrorln("{}\nwhile reading .npmrc \"{s}\"", .{
+                err,
+                ".npmrc",
+            });
+            Global.exit(1);
+        },
+    }
+    const source = bun.logger.Source.initPathString(".npmrc", npmrc_contents.items[0..]);
```

Comment:
Is there a reason we have another implementaiton of reading a file from disk, instead of using `bun.sys.File.toSource` or `bun.sys.File.toSourceAt`?